File: Collab_Backend/frontend/frontend_app/src/models/Assignment.ts
export class Assignment {
  id: string
  title: string
  description: string

  duedate: string
  estimatedTotalTime: number

  constructor(id: string, title: string, description: string, duedate: string, estimatedTotalTime: number) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.duedate = duedate;
    this.estimatedTotalTime = estimatedTotalTime;
  }
}

File: Collab_Backend/Gateway_App/src/main/java/com/backend/collab_backend/assignment/Assignment.java
package com.backend.collab_backend.assignment;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import java.time.LocalDate;
@Entity
@RequiredArgsConstructor
@Getter
@Setter
public class Assignment {
  @Id
  @SequenceGenerator(
          name="assignment_id_sequence",
          sequenceName = "assignment_id_sequence",
          allocationSize = 1
  )
  @GeneratedValue(
          strategy = GenerationType.SEQUENCE,
          generator = "assignment_id_sequence"
  )
  private Long assignmentId;

  @Column(nullable = false)
  private String course;
  @Column(nullable = false)
  private String title;

  @Column(nullable = false)
  private String description;

  @Column(nullable = false)
  private String groupId;
  @Column(nullable = false)
  private String type;
  @Column(nullable = false)
  private String time;
  @Column(nullable = false)
  private String teacherName;

  @Column(nullable = false)
  private LocalDate dueDate;

}